### PR TITLE
More granular debug

### PR DESCRIPTION
**Pull request by rebroad**
This pull request has been shunk down to size with other changes now to be in separate pull requests.

The change request here simply splits moves some of "net" debugging into "net2", "block" and "tx". 

This is a cut-down version of #8728 which also seeks to make debug message conform to a style.

Sorry for all the pull requests as a result of breaking this up, but I felt there was too much to it in it's previous format, and therefore was unlikely any of it would be merged.

### CODE REVIEW

**Comment by sipa** [comment]
Concept ACK

**Comment by MarcoFalke** [comment]
Wouldn't need init.cpp (debugCategories) an update as well?

**Comment by laanwj** [comment]
> Wouldn't need init.cpp (debugCategories) an update as well?

Yes, needs to be documented in option help.

**Comment by rebroad** [comment]
ok, will add a commit to update option help...  shall I list each debug string e.g. "tx, tx2, tx3"? or is it better to show the level of granularity another way, e.g. "tx{,2,3}" or "tx[1:3]", "tx[1-3]"...? not sure if there's a standard for denoting this sort of thing.

**Comment by rebroad** [comment]
@laanwj init.cpp now updated to show usage

File: src/init.cpp [hunk 1]
```diff
@@ -413,7 +413,7 @@ std::string HelpMessage(HelpMessageMode mode)
         strUsage += HelpMessageOpt("-limitdescendantsize=<n>", strprintf("Do not accept transactions if any ancestor would have more than <n> kilobytes of in-mempool descendants (default: %u).", DEFAULT_DESCENDANT_SIZE_LIMIT));
         strUsage += HelpMessageOpt("-bip9params=deployment:start:end", "Use given start/end times for specified BIP9 deployment (regtest-only)");
     }
-    string debugCategories = "addrman, alert, bench, coindb, db, http, libevent, lock, mempool, mempoolrej, net, proxy, prune, rand, reindex, rpc, selectcoins, tor, zmq"; // Don't translate these and qt below
+    string debugCategories = "addrman, alert, bench, block, coindb, db, estimatefee{,2}, http, libevent, lock, mempool(,2}, mempoolrej, net{,2}, proxy, prune, rand, reindex, rpc, selectcoins, tx{,2}, tor, zmq"; // Don't translate these and qt below
```
Comment (MarcoFalke) [inline src/init.cpp]: Can you also add `cmpctblock`?

Comment (MarcoFalke) [inline src/init.cpp]: μNit: I was wondering if it could help to add a "rule of thumb" somewhere to explain when to use `category` vs. `category2`.

Comment (rebroad) [inline src/init.cpp]: MarcoFalke, I propose to move the cmpctblock debugs into block{,2} since block is designed to cover all block downloads.

Comment (rebroad) [inline src/init.cpp]: @MarcoFalke will do so..  I propose that "cmpctblk" be restricted to the actual debug's relating to creation/etc of compact blocks, but for the transmission and download that block{,2} etc be used. Sound ok?

**Comment by rebroad** [comment]
On a slightly OT note - can someone tell me how one gets notifications when someone comments on a pull request? My currently practice is to scan through my pull requests to see if comments have appeared, but I suspect there may be a better way!!

**Comment by rebroad** [comment]
for some reason I can't re-open this :-s

**Comment by sipa** [comment]
@rebroad You already have another pull request from the same branch.

**Comment by rebroad** [comment]
still testing. will re-open when more done.

**Comment by sipa** [comment]
Can you please stop opening and closing this PR?

**Comment by MarcoFalke** [comment]
@rebroad Just leave this open. You can do as much testing as you want locally, but be sure to compile and run the tests before you push.

File: src/main.cpp [hunk 2]
```diff
@@ -4775,12 +4785,15 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                     CBlock block;
                     if (!ReadBlockFromDisk(block, (*mi).second, consensusParams))
                         assert(!"cannot load block from disk");
-                    if (inv.type == MSG_BLOCK)
+                    if (inv.type == MSG_BLOCK) {
+                        LogPrint(fRecent ? "block" : "block2", "sending regular %s (%d) to peer=%d\n", inv.ToString(), nHeight, pfrom->id);
```
Comment (n1bor) [inline src/main.cpp]: Any reason not to add the size of the block too? Useful to analyse bandwidth usage by block height.

Comment (rebroad) [inline src/main.cpp]: do you mean log the size when sending it or when receiving it, or both?

**Comment by rebroad** [comment]
Why is Travis complaining? 6 of the 7 travis tests were successful....

**Comment by paveljanik** [comment]
Click on Details... https://travis-ci.org/bitcoin/bitcoin/jobs/155868520#L1230

**Comment by rebroad** [comment]
This code is a bit of a mess at the moment, and much of the compact block stuff hasn't really been tested - very much WIP for those parts right now.

**Pull request closed without merging**